src/test/java/com/github/checkstyle/regression/module/ConfigGeneratorTest.java

```diff
@@ -0,0 +1,68 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import static com.github.checkstyle.regression.internal.TestUtils.assertUtilsClassHasPrivateConstructor;
+import static org.junit.Assert.assertEquals;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.github.checkstyle.regression.data.ConfigNode;
+import com.github.checkstyle.regression.data.GitChange;
+import com.github.checkstyle.regression.data.ImmutableConfigNode;
+
+public class ConfigGeneratorTest {
+    private static final String BASE_PACKAGE = "com.puppycrawl.tools.checkstyle";
+
+    private static final String JAVA_MAIN_SOURCE_PREFIX =
+            "src/main/java/com/puppycrawl/tools/checkstyle/";
+
+    @Test
+    public void testIsProperUtilsClass() throws Exception {
+        assertUtilsClassHasPrivateConstructor(ConfigGenerator.class);
+    }
+
+    @Test
+    public void testGenerateConfigNodes() {
```

Comment:
I split the current UT into two. Not very clear about "new invalid" mean. And what does "inside" mean?
Added UT:
```Java
@Test
    public void testGenerateConfigNodesForInvalidChanges() {
        final List<GitChange> changes = new LinkedList<>();
        changes.add(new GitChange(
                JAVA_MAIN_SOURCE_PREFIX + "PackageObjectFactory.java"));
        changes.add(new GitChange("foo/A.java"));
        final List<ConfigNode> configNodes = ConfigGenerator.generateConfigNodes(changes);
        assertEquals(0, configNodes.size());
    }
```
> a new invalid location inside `JAVA_MAIN_SOURCE_PREFIX`

I guess ` changes.add(new GitChange(JAVA_MAIN_SOURCE_PREFIX + "PackageObjectFactory.java"));` is this situation.

> inside `src\main\java`

Do you mean smth like `src/main/java/Foo.java`?